File: processor/src/test/resources/approvals/jsonb/LazyvalJsonbAdapters.java
package test;

import jakarta.json.bind.JsonbConfig;
import jakarta.json.bind.adapter.JsonbAdapter;
import scenarios.java.Ids;
import scenarios.java.Isbn;
import scenarios.java.OrderDate;
import scenarios.java.Quantity;

import javax.annotation.processing.Generated;
import java.time.LocalDate;

public class LazyvalJsonbAdapters {
  public static JsonbAdapter[] adapters() {
    return new JsonbAdapter[]{
            new IdsProductIdAdapter(),
            new IsbnAdapter(),
            new OrderDateAdapter(),
            new QuantityAdapter()
        };
  }

  public static JsonbConfig config() {
    return new JsonbConfig().withAdapters(adapters());
  }

  @Generated("com.qualityminds.lazyval.processor.internal.codegen.JsonbGenerator")
  static class IdsProductIdAdapter implements JsonbAdapter<Ids.ProductId, String> {
    @Override
    public String adaptToJson(Ids.ProductId obj) throws Exception {
      return obj.value();
    }

    @Override
    public Ids.ProductId adaptFromJson(String value) throws Exception {
      return Ids.ProductId.of(value);
    }
  }

  @Generated("com.qualityminds.lazyval.processor.internal.codegen.JsonbGenerator")
  static class IsbnAdapter implements JsonbAdapter<Isbn, String> {
    @Override
    public String adaptToJson(Isbn obj) throws Exception {
      return obj.getValue();
    }

    @Override
    public Isbn adaptFromJson(String value) throws Exception {
      return Isbn.parse(value);
    }
  }

  @Generated("com.qualityminds.lazyval.processor.internal.codegen.JsonbGenerator")
  static class OrderDateAdapter implements JsonbAdapter<OrderDate, LocalDate> {
    @Override
    public LocalDate adaptToJson(OrderDate obj) throws Exception {
      return obj.value();
    }

    @Override
    public OrderDate adaptFromJson(LocalDate value) throws Exception {
      return new OrderDate(value);
    }
  }

  @Generated("com.qualityminds.lazyval.processor.internal.codegen.JsonbGenerator")
  static class QuantityAdapter implements JsonbAdapter<Quantity, Integer> {
    @Override
    public Integer adaptToJson(Quantity obj) throws Exception {
      return obj.value();
    }

    @Override
    public Quantity adaptFromJson(Integer value) throws Exception {
      return new Quantity(value);
    }
  }
}

File: ksp/src/test/resources/approvals/jsonb/LazyvalJsonbAdapters.kt
package test

import jakarta.`annotation`.Generated
import jakarta.json.bind.JsonbConfig
import jakarta.json.bind.adapter.JsonbAdapter
import java.time.LocalDate
import kotlin.Array
import kotlin.Int
import kotlin.String
import kotlin.jvm.JvmStatic
import scenarios.kotlin.Ids
import scenarios.kotlin.Isbn
import scenarios.kotlin.NullableQuantity
import scenarios.kotlin.OrderDate
import scenarios.kotlin.Quantity

public class LazyvalJsonbAdapters {
  public companion object {
    @JvmStatic
    public fun adapters(): Array<JsonbAdapter<*, *>> = arrayOf(
      IdsProductIdAdapter(),
      IsbnAdapter(),
      NullableQuantityAdapter(),
      OrderDateAdapter(),
      QuantityAdapter()
    )

    @JvmStatic
    public fun config(): JsonbConfig = JsonbConfig().withAdapters(*adapters())
  }

  @Generated("com.qualityminds.lazyval.ksp.internal.codegen.JsonbGenerator")
  public class IdsProductIdAdapter : JsonbAdapter<Ids.ProductId, String> {
    override fun adaptToJson(obj: Ids.ProductId): String = obj.value

    override fun adaptFromJson(`value`: String): Ids.ProductId? = Ids.ProductId.of(value)
  }

  @Generated("com.qualityminds.lazyval.ksp.internal.codegen.JsonbGenerator")
  public class IsbnAdapter : JsonbAdapter<Isbn, String> {
    override fun adaptToJson(obj: Isbn): String = obj.value

    override fun adaptFromJson(`value`: String): Isbn? = Isbn.parse(value)
  }

  @Generated("com.qualityminds.lazyval.ksp.internal.codegen.JsonbGenerator")
  public class NullableQuantityAdapter : JsonbAdapter<NullableQuantity, Int> {
    override fun adaptToJson(obj: NullableQuantity): Int = obj.value

    override fun adaptFromJson(`value`: Int): NullableQuantity? = NullableQuantity.ofNullable(value)
  }

  @Generated("com.qualityminds.lazyval.ksp.internal.codegen.JsonbGenerator")
  public class OrderDateAdapter : JsonbAdapter<OrderDate, LocalDate> {
    override fun adaptToJson(obj: OrderDate): LocalDate = obj.date

    override fun adaptFromJson(`value`: LocalDate): OrderDate? = OrderDate(value)
  }

  @Generated("com.qualityminds.lazyval.ksp.internal.codegen.JsonbGenerator")
  public class QuantityAdapter : JsonbAdapter<Quantity, Int> {
    override fun adaptToJson(obj: Quantity): Int = obj.value

    override fun adaptFromJson(`value`: Int): Quantity? = Quantity(value)
  }
}
